reference/fuzz.go

```diff
@@ -0,0 +1,8 @@
+package reference
+
+// FuzzParseNormalizedNamed implements a fuzzer
+// that targets ParseNormalizedNamed
+func FuzzParseNormalizedNamed(data []byte) int {
```

Comment:
As `reference` is a package intended to be imported by others, I would prefer not to add `FuzzParseNormalizedNamed` as an exported function. Can you move this fuzz to its own package under an `internal` directory?

Reply:
I didn't find any existing `internal` directory, so instead of creating one for a single fuzzer, perhaps we could unexport the fuzzer, and when building it through OSS-fuzz we export it. This will work fine from the perspective of building it successfully through OSS-fuzz.
I did mention earlier that the fuzz tests need to be exported. This is only true when we build them, and they can easily remain unexported in their respective packages when we are not building them.

The current state of the files in the PR reflect the above suggestion. Let me know what you think.

Reply:
If that works then that's even nicer, thanks. Could we unexport the others?

Reply:
Done!

Reply:
@brackendawson Do you have a suggestion regarding the current failing CI test? The fuzzers are now not invoked.

```
reference/fuzz.go:8:6: `fuzzParseNormalizedNamed` is unused (deadcode)
func fuzzParseNormalizedNamed(data []byte) int {
     ^
configuration/fuzz.go:10:6: `parserFuzzer` is unused (deadcode)
func parserFuzzer(data []byte) int {
     ^
registry/api/v2/fuzz.go:7:6: `fuzzParseForwardedHeader` is unused (deadcode)
func fuzzParseForwardedHeader(data []byte) int {
```

Reply:
Oh jeez, very much looking forward to native fuzzing. For now just `// nolint;deadcode` them (https://golangci-lint.run/usage/false-positives/#nolint).